### PR TITLE
ProcessCapture: Refactor limiting of log output lines

### PR DESCRIPTION
**Pull request by sschuberth**
Output heading and trailing lines, as mostly errors occur at the end of
the output. Also share the same limiting code between generation of the
error message, and the debug logging.

Signed-off-by: Sebastian Schuberth <sebastian.schuberth@here.com>

---
This change is [Reviewable](https://reviewable.io/reviews/heremaps/oss-review-toolkit/1353)

### CODE REVIEW

File: utils/src/main/kotlin/ProcessCapture.kt [hunk 1]
```diff
@@ -35,16 +35,22 @@ class ProcessCapture(vararg command: String, workingDir: File? = null, environme
     constructor(workingDir: File?, vararg command: String) : this(*command, workingDir = workingDir)
 
     companion object {
-        private const val MAX_LOG_LINES = 20
-        private const val MAX_LOG_LINES_MESSAGE = "(Above output is limited to $MAX_LOG_LINES lines.)"
+        private const val MAX_OUTPUT_LINES = 20
+        private const val MAX_OUTPUT_FOOTER =
+                "(Above output is limited to $MAX_OUTPUT_LINES heading and tailing lines.)"
```
Comment (mnonnenmacher) [inline utils/src/main/kotlin/ProcessCapture.kt]: Maybe say `to each $MAX_OUTPUT_LINES heading and tailing lines`.

File: utils/src/main/kotlin/ProcessCapture.kt [hunk 2]
```diff
@@ -35,16 +35,22 @@ class ProcessCapture(vararg command: String, workingDir: File? = null, environme
...
+
+            return if (lines.count() > MAX_OUTPUT_LINES * 2) {
+                val prefix = lines.take(MAX_OUTPUT_LINES)
+                val suffix = lines.takeLast(MAX_OUTPUT_LINES)
 
-            // This actually already evaluates the next element, but since that is also only a chunk, not the whole
-            // rest, we can live with that.
-            if (chunkIterator.hasNext()) { logLine(MAX_LOG_LINES_MESSAGE) }
+                // Insert ellipsis in the middle of a long message.
+                (prefix + "[...]" + suffix + MAX_OUTPUT_FOOTER).joinToString("\n")
```
Comment (mnonnenmacher) [inline utils/src/main/kotlin/ProcessCapture.kt]: Maybe mention how many lines are skipped: `[...skipping $x lines...]`

File: utils/src/main/kotlin/ProcessCapture.kt (hunk 1 above)
Comment (sschuberth) [reply utils/src/main/kotlin/ProcessCapture.kt]: Done.

File: utils/src/main/kotlin/ProcessCapture.kt (hunk 2 above)
Comment (sschuberth) [reply utils/src/main/kotlin/ProcessCapture.kt]: Done.

**Pull request merged**